File: app/src/main/java/com/infy/dagger2implementation/database/dao/LocationDao.java
package com.infy.dagger2implementation.database.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.infy.dagger2implementation.database.model.LocationData;

import java.util.List;

@Dao
public interface LocationDao {
    @Query("SELECT * FROM locationdata")
    List<LocationData> getAll();

    @Query("SELECT * FROM locationdata WHERE latitue=:lat AND longitude=:lng")
    LocationData getByLatLng(String lat, String lng);

    @Insert
    void insertAll(List<LocationData> locationDataList);

    @Insert
    long insert(LocationData locationData);

    @Update
    void update(LocationData locationData);

    @Delete
    int delete(LocationData locationData);
}

File: app/src/main/java/com/infy/dagger2kotlin/database/dao/LocationDao.kt
package com.infy.dagger2kotlin.database.dao

import android.arch.persistence.room.*
import com.infy.dagger2kotlin.database.model.LocationData

@Dao
interface LocationDao {
    @get:Query("SELECT * FROM locationdata")
    val all: List<LocationData>

    @Query("SELECT * FROM locationdata WHERE latitue=:lat AND longitude=:lng")
    fun getByLatLng(lat: String, lng: String): LocationData

    @Insert
    fun insertAll(locationDataList: List<LocationData>)

    @Insert
    fun insert(locationData: LocationData): Long

    @Update
    fun update(locationData: LocationData)

    @Delete
    fun delete(locationData: LocationData): Int
}